python/metatensor-learn/metatensor/learn/data/collate.py

```diff
@@ -35,7 +35,11 @@ def group(batch: List[Tuple]) -> Tuple:
     return namedtuple("Batch", names)(*list(zip(*batch)))
 
 
-def group_and_join(batch: List[Tuple]) -> Tuple:
+def group_and_join(
+    batch: List[Tuple],
+    fields_to_join: Optional[List[str]] = None,
+    join_kwargs: Optional[dict] = {},
```

Comment:
```suggestion
    join_kwargs: Optional[dict] = None,
```

If you have optional, this feels more natural. Also, you should never use mutable values (i.e. empty dict/list) as default arguments: https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument